pytorch_lightning/plugins/training_type/tpu_spawn.py

```diff
@@ -312,7 +312,17 @@ def save_checkpoint(self, checkpoint: Dict[str, Any], filepath: _PATH) -> None:
             checkpoint: dict containing model and trainer state
             filepath: write-target file's path
         """
-        return self.checkpoint_io.save_checkpoint(checkpoint, filepath)
+        # `xla_model.save` needs to be called on all ranks. It internally checks if the local rank is 0
+        self.checkpoint_io.save_checkpoint(checkpoint, filepath)
+
+    def remove_checkpoint(self, filepath: _PATH) -> None:
+        """Remove checkpoint filepath from the filesystem.
+
+        Args:
+            filepath: Path to checkpoint
+        """
+        if self.local_rank == 0:
```

Comment:
This could be an issue in the future if the nodes filesystem becomes one a single volume.

Ideally, I think we should detect if the filesystem is shared. The condition would become (if self.global_rank if shared_filesystem else self.local_rank)

Reply:
I believe you have the same question as this comment: https://github.com/PyTorchLightning/pytorch-lightning/pull/11070/files#r770116930

Reply:
Yes, mind-opening an issue associated with this?